LLTextureCacheRemoteWorker::doRead Linux compilation error with -Werror=alloc-size-larger-than=

### Environment

Develop-Linux

### Description

PR with possible fix: https://github.com/secondlife/viewer/pull/5538

When compiling a Linux viewer where -Werror=alloc-size-larger-than= is set, the viewer will compilation fail [here](https://github.com/secondlife/viewer/blob/3dc67639d687669b4709ebeaedabc8f8af3c244b/indra/newview/lltexturecache.cpp#L349), since the compiler can't statically know if mDataSize can possibly be a negative value or not (it shouldn't ever be, but the compiler is unsure so it errors). 

> inlined from ‘virtual bool LLTextureCacheRemoteWorker::doRead()’ at /home/tj/secondlife-viewer/indra/newview/lltexturecache.cpp:349:46:
> /home/tj/secondlife-viewer/indra/llcommon/llmemory.h:175:23: error: argument 1 range [18446744071562067968, 18446744073709551615] exceeds maximum object size 9223372036854775807 [-Werror=alloc-size-larger-than=]

### Reproduction steps

1. Use a Linux build system
2. Ensure -Werror=alloc-size-larger-than= is enabled
3. Observe compilation fail doing mReadData = (U8*)ll_aligned_malloc_16(mDataSize) in LLTextureCacheRemoteWorker::doRead
<br/>
<hr/>

<details><summary>This repo is using Opire - what does it mean? 👇</summary><br/>💵 Everyone can add rewards for this issue commenting <code>/reward 100</code> (replace <code>100</code> with the amount).<br/>🕵️‍♂️ If someone starts working on this issue to earn the rewards, they can comment <code>/try</code> to let everyone know!<br/>🙌 And when they open the PR, they can comment <code>/claim #5537</code> either in the PR description or in a PR's comment.<br/><br/>🪙 Also, everyone can tip any user commenting <code>/tip 20 @Hecklezz</code> (replace <code>20</code> with the amount, and <code>@Hecklezz</code> with the user to tip).<br/><br/>📖 If you want to learn more, check out our <a href="https://docs.opire.dev">documentation</a>.</details>

--- a/indra/newview/lltexturecache.cpp
+++ b/indra/newview/lltexturecache.cpp
@@ -341,7 +341,7 @@ bool LLTextureCacheRemoteWorker::doRead()
     if (!done && (mState == LOCAL))
     {
         llassert(local_size != 0);  // we're assuming there is a non empty local file here...
-        if (!mDataSize || mDataSize > local_size)
+        if (mDataSize <= 0 || mDataSize > local_size)
         {
             mDataSize = local_size;
         }

